app/models/derived_variables/lettings_log_variables.rb

```diff
@@ -84,6 +84,15 @@ def set_derived_fields!
 
     if uprn_known&.zero?
       self.uprn = nil
+      if uprn_known_was == 1
+        self.address_line1 = nil
+        self.address_line2 = nil
+        self.town_or_city = nil
+        self.county = nil
+        self.postcode_known = nil
+        self.postcode_full = nil
+        self.la = nil
+      end
```

Comment:
If the uprn changed from known and selected to not known I think it should go back to what the values would have been if not know was selected to begin with